test/graphbuilder.cc

```diff
@@ -89,6 +90,40 @@ TEST(graphbuilder, TestConstructEdgesSubset) {
   EXPECT_TRUE(reader.DoesTileExist(GraphId{5993698}));
 }
 
+TEST(Graphbuilder, TestDEBuilderLength) {
+
+  std::vector<PointLL> shape1{{-160.096619f, 21.997619f},
+                              {-90.037697f, 41.004531},
+                              {-160.096619f, 21.997619f}};
+  try {
+    DirectedEdgeBuilder edge_builder({}, GraphId(123, 2, 8), true, valhalla::midgard::length(shape1),
+                                     1, 1, Use::kRoad, baldr::RoadClass::kMotorway, 0, false, 0, 0,
+                                     false);
+
+  } catch (...) {
+    // should not throw an exception.
+    throw std::runtime_error(
+        "TestDEBuilderLength: DirectedEdgeBuilder: Should not have thrown an exception for the length");
+  }
+
+  std::vector<PointLL> shape2{{-160.096619f, 21.997619f},
+                              {-90.037697f, 41.004531},
+                              {-160.096619f, 21.997619f},
+                              {-90.037697f, 41.004531}};
+  try {
+    DirectedEdgeBuilder edge_builder({}, GraphId(123, 2, 8), true, valhalla::midgard::length(shape2),
```

Comment:
Consider `ASSERT_THROW`/`EXCEPT_THROW`

Reply:
sure.  will do

Reply:
hmmm...won't assert stop execution? Since this is during data building we can perhaps expect times when a new OSM way / edge is added that might exceed this (e.g. long ferries). I was reading that asserts should be used when you know some condition that should never occur - here is a condition that could occur and I think should be an exception.